addons/portal/controllers/mail.py

```diff
@@ -112,6 +112,22 @@ def _portal_post_has_content(self, res_model, res_id, message, attachment_ids=No
         """ Tells if we can effectively post on the model based on content. """
         return bool(message) or bool(attachment_ids)
 
+    @http.route('/mail/avatar/mail.message/<int:res_id>/author_avatar/<int:width>x<int:height>', type='http', auth='public')
+    def portal_avatar(self, res_id=None, height=50, width=50, access_token=None):
+        """ Get the avatar image in the chatter of the portal """
+        Message = request.env['mail.message']
+
+        # Set default image if there is no token
+        record = request.env.ref('web.image_placeholder').sudo()
+        stream = request.env['ir.binary']._get_image_stream_from(record, width=int(width), height=int(height))
+        if access_token:
+            record = Message.browse(res_id)
+            access_as_sudo = _check_special_access(record.model, record.res_id, token=access_token)
+            if access_as_sudo:
+                stream = request.env['ir.binary']._get_image_stream_from(record, 'author_avatar', width=int(width), height=int(height))
```

Comment:
Currently this works for the same reason it worked in 15.0
But it would be good to sudo the record to explicitly show we won't be checking access rules/rights